test/Test/Pos/Block/Logic/Mode.hs

```diff
@@ -136,22 +138,22 @@ instance Buildable TestParams where
             _tpStakeDistribution
             _tpStartTime
       where
-        utxo =  unGenesisUtxo (_tpGenTxpContext ^. gtcUtxo)
+        utxo = unGenesisUtxo (_tpGenesisContext ^. gtcUtxo)
 
 instance Show TestParams where
     show = formatToString build
 
 -- More distributions can be added if we want (e. g. RichPoor).
-genSuitableStakeDistribution :: Word -> Gen StakeDistribution
-genSuitableStakeDistribution stakeholdersNum =
-    oneof [genFlat{-, genBitcoin-}, pure (ExponentialStakes stakeholdersNum)]
+genSuitableStakeDistribution :: Coin -> Word -> Gen StakeDistribution
+genSuitableStakeDistribution dustThd stakeholdersNum =
+    -- Minimum coin in distribution should be equal or greater than
+    -- boot dust threshold.
+    stakeDistrMapCoin (`unsafeAddCoin` dustThd) <$>
+    oneof [ genFlat {-, genBitcoin-} -- is broken
+          , pure (ExponentialStakes stakeholdersNum (mkCoin 0))]
```

Comment:
How does `ExponentialStakes` work with `mkCoin 0`? IIRC, there is `iterate (* 2)`, so everyone should get 0 😕.